docs/flatcar-linux/kvm-libvirt.md

```diff
@@ -0,0 +1,349 @@
+* At least 4 GB of free RAM
+* Running libvirtd system service, see [libvirtd Setup](#libvirtd-setup)
+* A user in the `libvirt` group (to create virtual machines with libvirt and access `/dev/kvm`), see [User Setup]
+* Terraform v0.11.x, [terraform-provider-ct](https://github.com/poseidon/terraform-provider-ct),
+  and [terraform-provider-libvirt](https://github.com/dmacvicar/terraform-provider-libvirt)
+  installed locally
+
+## Prepare VM Image
+
+Download the Flatcar image. The example below uses the "Edge" Flatcar channel.
+You need to increase its size depending on your use case because the initial size
+is only enough for some small pods to be loaded. Starting pods with normal-sized
+container images may already fail because the disk space will be filled.
+That's why the example below directly increases the image size by 5 GB to make
+sure that a couple of container images fit on the node.
+
+```sh
+$ sudo dnf install wget bzip2 qemu-img  # or sudo apt install wget bzip2 qemu-utils
+$ wget https://edge.release.flatcar-linux.net/amd64-usr/2135.99.0/flatcar_production_qemu_image.img.bz2
+$ bunzip2 flatcar_production_qemu_image.img.bz2
+$ qemu-img resize flatcar_production_qemu_image.img +5G
+```
+
+## libvirtd Setup
+
+Ensure that libvirtd is set up.
+The examples in this guide use Fedora's `dnf`
+but for example in Ubuntu you would use `apt` as stated in the comment in the same line.
+
+```sh
+$ sudo dnf install libvirt-daemon  # or sudo apt install libvirt-daemon
+$ sudo systemctl start libvirtd
+```
+
+## User Setup
+
+Ensure that libvirt is accessible for the current user:
+
+```sh
+$ sudo usermod -a -G libvirt $(whoami)
+$ newgrp libvirt
+```
+
+## Terraform Setup
+
+Install [Terraform](https://www.terraform.io/downloads.html) v0.11.x on your system.
+
+```sh
+$ terraform version
+Terraform v0.11.13
+```
+
+Add the [terraform-provider-ct](https://github.com/poseidon/terraform-provider-ct) plugin binary for your system
+to `~/.terraform.d/plugins/`, noting the `_v0.3.1` suffix.
+
+```sh
+wget https://github.com/poseidon/terraform-provider-ct/releases/download/v0.3.1/terraform-provider-ct-v0.3.1-linux-amd64.tar.gz
+tar xzf terraform-provider-ct-v0.3.1-linux-amd64.tar.gz
+mv terraform-provider-ct-v0.3.1-linux-amd64/terraform-provider-ct ~/.terraform.d/plugins/terraform-provider-ct_v0.3.1
+```
+
+Add the [terraform-provider-libvirt](https://github.com/dmacvicar/terraform-provider-libvirt) plugin binary for your system
+to `~/.terraform.d/plugins/`, noting the `_v0.5.2` suffix.
+You should use the 0.5.2 release because it includes the experimental
+[pool definition feature](https://github.com/dmacvicar/terraform-provider-libvirt/commit/9f00f3d46c489f24c71d02fde816f5fda34d3f7c)
+but building from source is also an option.
+
+```sh
+wget https://github.com/dmacvicar/terraform-provider-libvirt/releases/download/v0.5.2/terraform-provider-libvirt-0.5.2.Fedora_28.x86_64.tar.gz
+tar xzf terraform-provider-libvirt-0.5.2.Fedora_28.x86_64.tar.gz
```

Comment:
Isn't this Fedora specific?

Reply:
Rather than Fedora maybe say <Distro> or tell the user that this is Fedora specific, referring them to the releases page where they can find their distro

Reply:
done